lib/jekyll-admin/server/collection.rb

```diff
@@ -59,14 +65,44 @@ def document
         collection.docs.find { |d| d.path == document_path }
       end
 
+      def directory_docs
+        collection.docs.find_all { |d| File.dirname(d.path) == directory_path }
+      end
+
+      def directory_path
+        sanitized_path File.join(collection.relative_directory, params["splat"].first)
+      end
+
+      def draft_path
+        sanitized_path File.join(site.source, '_drafts')
+      end
+
       def ensure_collection
         render_404 if collection.nil?
       end
 
+      def ensure_directory
+        ensure_collection
+        render_404 unless Dir.exists?(directory_path)
+      end
+
       def ensure_document
         ensure_collection
         render_404 if document.nil?
       end
+
+      def entries
+        args = {
+          base: site.source,
+          content_type: params['collection_id'],
+          splat: params["splat"].first
```

Comment:
I need this to fill the gap in the `api_url`. For example, if the requested page in `test/other` directory relative to `pages`, I need to be able to generate the `api_url` like this;

/_api/pages/entries/**test/other**/test.md